src/data/int/basic.lean

```diff
@@ -794,22 +795,20 @@ lemma units_inv_eq_self (u : units ℤ) : u⁻¹ = u :=
 @[simp] lemma bodd_two : bodd 2 = ff := rfl
 
 @[simp] lemma bodd_sub_nat_nat (m n : ℕ) : bodd (sub_nat_nat m n) = bxor m.bodd n.bodd :=
-by apply sub_nat_nat_elim m n (λ m n i, bodd i = bxor m.bodd n.bodd);
-   intros i m; simp [bodd]; cases i.bodd; cases m.bodd; refl
+by apply sub_nat_nat_elim m n (λ m n i, bodd i = bxor m.bodd n.bodd); intros;
+  simp [bodd, -of_nat_eq_coe]
```

Comment:
I’m a bit concerned about adding a simp lemma where we immediately have to exclude it multiple times. What happens in these cases if you leave it in?

Reply:
We have to decide what we want the simp normal form to for an natural number mapped to an integer, `(n : int)` or `int.of_nat n`. Currently `simp` doesn't recognize that they are equal, which is bad.
I think most lemmas assume that the former is the simp normal form, so I went with that. I'm not aware of any lemma that assume that the latter is the simp normal form (though there might be some). However, some proofs (like these ones I had to fix) add non-simp lemmas to their simp-set which try to normalize `int.of_nat n`. I don't think that the existence of such proofs should discourage us from adding this simp-lemma. 
A slightly more clean way to fix these proofs is to use different rewrite lemmas (that do not try to normalize `int.of_nat n`), but to be honest I don't care that much about the quality of these proofs to do that. 
The fact that these proofs are crappy can be realized from the fact that they were proving the lemmas I added to `bool` instead of factoring them out.

Reply:
Sounds good to me.